src/geventhttpclient/client.py

```diff
@@ -156,9 +157,12 @@ def request(self, method, request_uri, body=b"", headers={}):
         while 1:
             sock = self._connection_pool.get_socket()
             try:
-                sock.sendall(request)
+                sock.sendall(request.encode())
                 if body:
-                    sock.sendall(body)
+                    if type(body) == six.text_type:
+                        sock.sendall(body.encode('iso-8859-1'))
```

Comment:
six.text_type is unicode() in Python2 and and str in Python3. 

I wanted to follow the official client more closely. When looking at [HTTPConnection.request](https://docs.python.org/3/library/http.client.html#http.client.HTTPConnection.request) it does exactly that. It would theoretically even allow files.

It says that if something which is not b'' is given, it will be encoded with 'iso-8859-1'.